### PR TITLE
Rewrite syntax highlighting with semantic token provider

### PR DESCRIPTION
**Pull request by JamesPHoughton**
## Summary

Replaces the TextMate-only grammar approach (which couldn't layer domain tokens on top of YAML) with a two-pass system:

1. **TextMate grammar** (instant) — minimal, inherits `source.yaml` + `${templateVar}` highlighting
2. **Semantic token provider** (TypeScript) — walks the YAML AST and assigns standard VS Code token types based on structural context

### Token assignments

| Content | Token type | Theme color (Dark+) |
|---|---|---|
| Element types (`prompt`, `submitButton`, etc.) | `type` | Teal |
| Comparator values (`equals`, `isAbove`, etc.) | `keyword` | Purple |
| Reference strings (`prompt.q1`, `survey.name.path`) | `variable` | Blue |
| File paths | `string` | Yellow |
| Section keys (`elements`, `treatments`, `conditions`) | `property` | Blue |
| contentType values (`elements`, `stage`, etc.) | `type` | Teal |
| Separator styles (`thin`, `thick`, `regular`) | `keyword` | Purple |
| Position/chatType enums (`shared`, `player`, `all`) | `keyword` | Purple |

### Schema-driven

Element types, comparators, and reference types are imported from `stagebook` (`validElementTypes`, `validComparators`, `validReferenceTypes`). Adding a new element type to the schema automatically updates the highlighting.

### Approach change from issue spec

Issue #77 originally asked for a TextMate grammar rewrite. After iterating, we found that TextMate grammars fundamentally cannot layer domain tokens on top of another grammar's tokenization without copying the entire grammar. The semantic token approach is superior: it parses the YAML AST (which we already do for validation) and assigns tokens based on actual structural context, not regex heuristics.

Refs #77

## Test plan
- [x] 21 semantic token tests + 67 existing = 87 total
- [x] Negative tests: invalid comparators, invalid references, invalid contentType
- [x] Edge cases: empty input, comment-only YAML
- [x] All monorepo tests pass
- [x] Manual testing: installed VSIX and verified colors in real treatment files

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: apps/vscode/syntaxes/treatmentsYaml.tmLanguage.json [hunk 1]
```diff
@@ -2,15 +2,66 @@
...
+    "comparators": {
+      "comment": "Condition comparator values — all 16 from conditionSchema",
+      "match": "(?<=:\\s)\\b(exists|doesNotExist|equals|doesNotEqual|isAbove|isBelow|isAtLeast|isAtMost|hasLengthAtLeast|hasLengthAtMost|includes|doesNotInclude|matches|doesNotMatch|isOneOf|isNotOneOf)\\b",
+      "name": "keyword.control.comparator.treatments-yaml"
+    },
+
+    "element-type-value": {
+      "comment": "Element type values — matched after 'type:' to avoid highlighting these words in other contexts",
+      "match": "(?<=type:\\s)\\b(audio|image|display|prompt|separator|sharedNotepad|submitButton|survey|talkMeter|timer|mediaPlayer|timeline|qualtrics|trackedLink)\\b",
+      "name": "entity.name.type.element.treatments-yaml"
```
[the remark this answers is not in the document]
Comment (JamesPHoughton) [reply apps/vscode/syntaxes/treatmentsYaml.tmLanguage.json]: These TextMate patterns were removed in the subsequent commit — the grammar is now minimal (`source.yaml` + template variables only). All domain-specific highlighting is handled by the semantic token provider in `semanticTokens.ts`, which walks the YAML AST with full structural context, avoiding the false-positive issues described here.

File: apps/vscode/syntaxes/treatmentsYaml.tmLanguage.json [hunk 2]
```diff
@@ -2,15 +2,66 @@
...
+      "comment": "Reference namespace prefixes in reference strings (e.g. 'prompt.elementName')",
+      "match": "\\b(prompt|survey|discussion|participantInfo|urlParams|submitButton|qualtrics|trackedLink|connectionInfo|browserInfo)\\.",
+      "captures": {
+        "1": { "name": "support.function.reference.treatments-yaml" }
+      }
+    },
+
+    "separator-style": {
+      "comment": "Separator style enum values",
+      "match": "(?<=style:\\s)\\b(thin|thick|regular)\\b",
```
[the remark this answers is not in the document]
Comment (JamesPHoughton) [reply apps/vscode/syntaxes/treatmentsYaml.tmLanguage.json]: These TextMate patterns were removed in the subsequent commit — the grammar is now minimal (`source.yaml` + template variables only). All domain-specific highlighting is handled by the semantic token provider in `semanticTokens.ts`, which walks the YAML AST with full structural context, avoiding the false-positive issues described here.

File: apps/vscode/syntaxes/treatmentsYaml.tmLanguage.json [hunk 3]
```diff
@@ -2,15 +2,66 @@
...
+      "name": "constant.other.separator-style.treatments-yaml"
+    },
+
+    "domain-enum-value": {
+      "comment": "Domain-specific enum values: position selectors and chat types",
+      "match": "(?<=:\\s)\\b(shared|player|all|any|percentAgreement|text|audio|video)\\b",
```
[the remark this answers is not in the document]
Comment (JamesPHoughton) [reply apps/vscode/syntaxes/treatmentsYaml.tmLanguage.json]: These TextMate patterns were removed in the subsequent commit — the grammar is now minimal (`source.yaml` + template variables only). All domain-specific highlighting is handled by the semantic token provider in `semanticTokens.ts`, which walks the YAML AST with full structural context, avoiding the false-positive issues described here.

File: apps/vscode/syntaxes/treatmentsYaml.tmLanguage.json [hunk 4]
```diff
@@ -2,15 +2,66 @@
...
+    "element-type-value": {
+      "comment": "Element type values — matched after 'type:' to avoid highlighting these words in other contexts",
+      "match": "(?<=type:\\s)\\b(audio|image|display|prompt|separator|sharedNotepad|submitButton|survey|talkMeter|timer|mediaPlayer|timeline|qualtrics|trackedLink)\\b",
+      "name": "entity.name.type.element.treatments-yaml"
+    },
+
+    "top-level-section-key": {
+      "comment": "Structural section keys in treatment files",
+      "match": "(?<=^\\s*)\\b(introSequences|introSteps|gameStages|elements|treatments|templates|exitSequence|groupComposition|templateName|templateContent|contentType|broadcast)\\b(?=\\s*:)",
+      "name": "support.class.section.treatments-yaml"
```
[the remark this answers is not in the document]
Comment (JamesPHoughton) [reply apps/vscode/syntaxes/treatmentsYaml.tmLanguage.json]: These TextMate patterns were removed in the subsequent commit — the grammar is now minimal (`source.yaml` + template variables only). All domain-specific highlighting is handled by the semantic token provider in `semanticTokens.ts`, which walks the YAML AST with full structural context, avoiding the false-positive issues described here.

File: apps/vscode/syntaxes/treatmentsYaml.tmLanguage.json [hunk 5]
```diff
@@ -2,15 +2,66 @@
...
+    },
+
+    "condition-key": {
+      "comment": "Keys within condition blocks",
+      "match": "(?<=^\\s*)\\b(conditions|comparator|reference|position)\\b(?=\\s*:)",
+      "name": "keyword.operator.condition.treatments-yaml"
```
[the remark this answers is not in the document]
Comment (JamesPHoughton) [reply apps/vscode/syntaxes/treatmentsYaml.tmLanguage.json]: These TextMate patterns were removed in the subsequent commit — the grammar is now minimal (`source.yaml` + template variables only). All domain-specific highlighting is handled by the semantic token provider in `semanticTokens.ts`, which walks the YAML AST with full structural context, avoiding the false-positive issues described here.

File: apps/vscode/syntaxes/treatmentsYaml.tmLanguage.json [hunk 6]
```diff
@@ -2,15 +2,66 @@
...
   "repository": {
     "template-variable": {
       "match": "\\$\\{[a-zA-Z0-9_]+\\}",
       "name": "variable.other.constant.treatments-yaml"
+    },
+
+    "comparators": {
+      "comment": "Condition comparator values — all 16 from conditionSchema",
+      "match": "(?<=:\\s)\\b(exists|doesNotExist|equals|doesNotEqual|isAbove|isBelow|isAtLeast|isAtMost|hasLengthAtLeast|hasLengthAtMost|includes|doesNotInclude|matches|doesNotMatch|isOneOf|isNotOneOf)\\b",
+      "name": "keyword.control.comparator.treatments-yaml"
```
[the remark this answers is not in the document]
Comment (JamesPHoughton) [reply apps/vscode/syntaxes/treatmentsYaml.tmLanguage.json]: These TextMate patterns were removed in the subsequent commit — the grammar is now minimal (`source.yaml` + template variables only). All domain-specific highlighting is handled by the semantic token provider in `semanticTokens.ts`, which walks the YAML AST with full structural context, avoiding the false-positive issues described here.
